java/src/main/java/vowpalWabbit/learner/VWLearnerBase.java

```diff
@@ -17,7 +17,7 @@
  *
  * @author deak
  */
-abstract class VWLearnerBase<T> extends VWBase implements VWTypedLearner<T> {
+public abstract class VWLearnerBase<T> extends VWBase implements VWTypedLearner<T> {
```

Comment:
**LOOK AT ME**.  I still don't know about this.  Do we really need to do this?  I think that we could make a public marker interface instead or something.  I don't think we should make public the abstract classes, only the interfaces.